src/Compilers/CSharp/Test/Semantic/Semantics/DelegateTypeTests.cs

```diff
@@ -11494,5 +11495,671 @@ .maxstack  0
+        {
+            CheckNames(new[] { reader }, handles, expectedNames);
+        }
+
+
+        [Fact]
+        public void LambdaWithDefaultParameter()
+        {
+            var source = """
+using System;
+
+class Program
+{
+    // static void Report(object d) => Console.WriteLine(d.GetType());
+    public static void Main()
+    {   
+        var lam = (int x = 30) => x;
+        Console.WriteLine(lam() + " " + lam(10));
+       // Report(lam);
+    } 
+}
+""";
+
+            var expectAnonymousDelegateIL = """
+.class private auto ansi sealed '<>f__AnonymousDelegate0'
+	extends [netstandard]System.MulticastDelegate
+{
+	.custom instance void [netstandard]System.Runtime.CompilerServices.CompilerGeneratedAttribute::.ctor() = (
+		01 00 00 00
+	)
+	// Methods
+	.method public hidebysig specialname rtspecialname 
+		instance void .ctor (
+			object 'object',
+			native int 'method'
+		) runtime managed 
+	{
+	} // end of method '<>f__AnonymousDelegate0'::.ctor
+	.method public hidebysig newslot virtual 
+		instance int32 Invoke (
+			[opt] int32 ''
+		) runtime managed 
+	{
+		.param [1] = int32(30)
+	} // end of method '<>f__AnonymousDelegate0'::Invoke
+} // end of class <>f__AnonymousDelegate0
+""";
+
+
+            var expectLoweredClosureContainerIL = """
+.class nested private auto ansi sealed serializable beforefieldinit '<>c'
+	extends [netstandard]System.Object
+{
+	.custom instance void [netstandard]System.Runtime.CompilerServices.CompilerGeneratedAttribute::.ctor() = (
+		01 00 00 00
+	)
+	// Fields
+	.field public static initonly class Program/'<>c' '<>9'
+	.field public static class '<>f__AnonymousDelegate0' '<>9__0_0'
+	// Methods
+	.method private hidebysig specialname rtspecialname static 
+		void .cctor () cil managed 
+	{
+		// Method begins at RVA 0x20b4
+		// Code size 11 (0xb)
+		.maxstack 8
+		IL_0000: newobj instance void Program/'<>c'::.ctor()
+		IL_0005: stsfld class Program/'<>c' Program/'<>c'::'<>9'
+		IL_000a: ret
+	} // end of method '<>c'::.cctor
+	.method public hidebysig specialname rtspecialname 
+		instance void .ctor () cil managed 
+	{
+		// Method begins at RVA 0x20ac
+		// Code size 7 (0x7)
+		.maxstack 8
+		IL_0000: ldarg.0
+		IL_0001: call instance void [netstandard]System.Object::.ctor()
+		IL_0006: ret
+	} // end of method '<>c'::.ctor
+	.method assembly hidebysig 
+		instance int32 '<Main>b__0_0' (
+			[opt] int32 x
+		) cil managed 
+	{
+		.param [1] = int32(30)
+		// Method begins at RVA 0x20c0
+		// Code size 2 (0x2)
+		.maxstack 8
+		IL_0000: ldarg.1
+		IL_0001: ret
+	} // end of method '<>c'::'<Main>b__0_0'
+} // end of class <>c
+""";
+
+            var verifier = CompileAndVerify(source, expectedOutput: "30 10");
+            var m = ModuleMetadata.CreateFromImage(verifier.EmittedAssemblyData);
+            var reader = m.MetadataReader;
+            CheckNames(reader, reader.GetTypeDefNames(), "<Module>", "<>f__AnonymousDelegate0", "Program", "<>c");
```

Comment:
Thankfully, none of this manual module reading is necessary. `CompileAndVerify` has a `symbolVerifier` callback that will be passed the `ModuleSymbol` instance after being compiled and re-read. However, I don't think any of this is necessary: verifying the type IL seems fine to me.

Another option, instead of verifying the type IL, would be to use `symbolVerifier` to grab these symbols from the module, and verify the properties about them that we want (the invoke method has the default parameter in the right place, for example).